### PR TITLE
Separated Balances and Prices Calls

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Separated Balances and Prices Calls

Before we would call for `Prices` and `Balances` in the same method and the `Balances` was dependent on `Prices` response.
They are now separated into 2 different methods `refreshBalances` and `refreshPrices`.
This will now allow `Balances` to be displayed to the user even if the `Prices` haven't been returned yet.

(note: In the video I intentionally delayed prices from being fetched for `10` seconds to give an example.)

Resolves <https://github.com/brave/brave-browser/issues/19407>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

https://user-images.githubusercontent.com/40611140/143087786-b9570a60-88db-40fe-b2d2-9d24c102ed74.mov

### CODE REVIEW

File: components/brave_wallet_ui/page/container.tsx [hunk 1]
```diff
@@ -291,7 +291,7 @@ function Container (props: Props) {
       let fiatBalance
       const found = account.tokens.find((token) => token.asset.contractAddress === asset.contractAddress)
       if (found) {
-        fiatBalance = Number(found.fiatBalance)
+        fiatBalance = found.fiatBalance !== '' ? Number(found.fiatBalance) : undefined
```
Comment (onyb) [inline components/brave_wallet_ui/page/container.tsx:294]: Maybe you could do something like this:

```ts
if (found && found.fiatBalance !== '') {
  fiatBalance = Number(found.fiatBalance)
}
```

File: components/brave_wallet_ui/common/reducers/wallet_reducer.ts [hunk 2]
```diff
@@ -165,10 +165,11 @@ reducer.on(WalletActions.nativeAssetBalancesUpdated, (state: any, payload: GetNa
   accounts.forEach((account, index) => {
     if (payload.balances[index].success) {
       accounts[index].balance = payload.balances[index].balance
-      accounts[index].fiatBalance = formatFiatBalance(payload.balances[index].balance, state.selectedNetwork.decimals, payload.usdPrice).toString()
+      accounts[index].fiatBalance = payload.usdPrice !== ''
+        ? formatFiatBalance(payload.balances[index].balance, state.selectedNetwork.decimals, payload.usdPrice).toString()
+        : ''
```
Comment (onyb) [inline components/brave_wallet_ui/common/reducers/wallet_reducer.ts:170]: Does it make sense to use the cached `usdPrice` from the state? Also, what do you think about `payload.usdPrice !== undefined` here? Unless the `usdPrice` is supposed to be rendered as an empty string somewhere, I think it's better to keep the `''` concept specific to unknown balances.

**Review by onyb — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/page/container.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/container.tsx]: Updated!

File: components/brave_wallet_ui/common/reducers/wallet_reducer.ts (hunk 2 above)
Comment (onyb) [reply components/brave_wallet_ui/common/reducers/wallet_reducer.ts:170]: Discussed internally - it's a non-issue.

**Review by onyb — APPROVED** [review]

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.34.46 Chromium: 96.0.4664.45 (Official Build) nightly (x86_64 translated)
-- | --
Revision | `76e4c1bb2ab4671b8beba3444e61c0f17584b2fc-refs/branch-heads/4664@{#947}`
OS | macOS Version 12.0.1 (Build 21A559)

- Verified balance is shown when price value fetch is delayed
- Verified balance and price calls are independent of each other

https://user-images.githubusercontent.com/17010094/144010126-97e65ceb-84e4-44ac-8d23-4054cd66716a.mov

**Pull request merged**